sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -92,6 +107,20 @@
  * PCollection<String> lines = filenames.apply(TextIO.readAll());
  * }</pre>
  *
+ * <p>Example 3: streaming new files matching a filepattern.
+ *
+ * <pre>{@code
+ * Pipeline p = ...;
+ *
+ * PCollection<String> lines = p.apply(TextIO.read()
+ *     .from("/local/path/to/files/*")
```

Comment:
I think so - I added the Match transform to support this. Everything looks cleaner now.